### ISSUE TITLE
scalable watcher

### ISSUE THREAD

**Issue by gedw99** [infogulch/watch#1]
I don't know if this helps but I have had a good experience using https://github.com/cosmtrek/air/blob/master/runner/watcher.go#L6

its building on the should of Hugo

**Comment by infogulch** [comment]
Thanks for the reference, I wasn't aware of Hugo's implementation.

I've had a good experience with fsnotify library, which is nice because it doesn't do any polling. Have you tried fsnotify as well?

**Comment by gedw99** [comment]
yep I have used it but I found it has some problems in terms of cross platform like on windows, and so I used the Hugo wrapper.
